### PR TITLE
Gutenboarding: Remove click-focus style from viewports

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

Remove click-initiated focus style on buttons.
If buttons are focused by keyboard navigation, maintain focus style.

Part of #40757

##### Before
Screen Shot 2020-04-06 at 12 31 32

##### After
Screen Shot 2020-04-06 at 12 31 46

#### Testing instructions

* [`/gutenboarding`](/https://calypso.live/gutenboarding?branch=gutenboarding/remove-viewport-active-border)
* Proceed to style step
* Click different viewports
* On hover they should have a box
* The last clicked (focused) button should not
* If focused via keyboard navigation, it should have focus styles regardless (has box)

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/style-preview/style.scss [hunk 1]
```diff
@@ -160,24 +160,36 @@ $gutenboarding-style-preview-bar-height: 30px;
...
-		&.is-selected {
-			color: var( --studio-black );
-		}
+	&.is-selected {
+		color: var( --studio-black );
 	}
 }
 
+// Remove focus styling from clicking a button
+// Keep keyboard-focused focus styl
```
Comment (simison) [inline client/landing/gutenboarding/onboarding-block/style-preview/style.scss:186]: `styl` -> `style`

File: client/landing/gutenboarding/onboarding-block/style-preview/style.scss [hunk 2]
```diff
@@ -160,24 +160,36 @@ $gutenboarding-style-preview-bar-height: 30px;
...
+html:not( .accessible-focus )
+	.style-preview__viewport-select-button.is-selected:focus:not( :hover ) {
+	// outline: 1px dotted red !important;
```
Comment (simison) [inline client/landing/gutenboarding/onboarding-block/style-preview/style.scss:189]: commented out code

File: client/landing/gutenboarding/onboarding-block/style-preview/style.scss (hunk 1 above)
Comment (sirreal) [reply client/landing/gutenboarding/onboarding-block/style-preview/style.scss:186]:
```suggestion
// Keep keyboard-focused focus style
```

File: client/landing/gutenboarding/onboarding-block/style-preview/style.scss (hunk 2 above)
Comment (sirreal) [reply client/landing/gutenboarding/onboarding-block/style-preview/style.scss:189]:
```suggestion
```

**Comment by yansern** [comment]
@sirreal 
✅On hover they should have a box
❌The last clicked (focused) button should not
✅If focused via keyboard navigation, it should have focus styles regardless (has box)

Need clarity here, the last clicked (focused) button should not have a box **when hovered** right?

**Comment by sirreal** [comment]
> Need clarity here, the last clicked (focused) button should not have a box **when hovered** right?

Hover _always_ gets a box.

**Pull request merged**